cli/tests/integrations/test_cluster.py

```diff
@@ -29,6 +34,58 @@ def test_list():
     assert sorted(info.keys()) == keys
 
 
+@pytest.fixture
+def cluster_backup():
+    dcos_dir = os.environ.get('DCOS_DIR')
+    if dcos_dir is None:
+        dcos_dir = dcos.config.get_config_dir_path()
+    assert dcos_dir is not None
+    cluster_dir = os.path.join(dcos_dir, 'clusters')
+    back_dir = os.path.join(dcos_dir, 'backup')
+    copy_tree(cluster_dir, back_dir)
+
+    yield cluster_dir
+    # return to order
+    remove_tree(cluster_dir)
+    copy_tree(back_dir, cluster_dir)
+    remove_tree(back_dir)
```

Comment:
Even though you seem to clean it up properly, I wouldn't use the actual dcos dir as a fixture, this can leave subtle inconsistencies or become a pain to debug if something goes wrong (one would have to remove the folder, setup the cluster again to start from a clean state).

Could you instead copy the whole dcos dir to a temporary location, make your changes and then set it as `DCOS_DIR`, there is something similar here (except that it doesn't copy the current one) : https://github.com/dcos/dcos-cli/blob/ce504929eadbb60e3befeeb03a65566d535482bb/cli/tests/integrations/test_cluster_migration.py#L18-L26

Reply:
this is better!